src/gui/FileBrowser.cpp

```diff
@@ -629,37 +630,63 @@ void FileBrowserTreeWidget::focusOutEvent(QFocusEvent* fe)
 
 void FileBrowserTreeWidget::contextMenuEvent(QContextMenuEvent * e )
 {
-	auto file = dynamic_cast<FileItem*>(itemAt(e->pos()));
-	if( file != nullptr && file->isTrack() )
-	{
-		QMenu contextMenu( this );
+#ifdef LMMS_BUILD_APPLE
+	QString fileManager = tr("Finder");
+#elif defined(LMMS_BUILD_WIN32)
+	QString fileManager = tr("Explorer");
+#else
+	QString fileManager = tr("file manager");
+#endif
 
-		contextMenu.addAction(
-			tr( "Send to active instrument-track" ),
-			[=, this]{ sendToActiveInstrumentTrack(file); }
-		);
+   QTreeWidgetItem* item = itemAt(e->pos());
 
-		contextMenu.addSeparator();
+   auto file = dynamic_cast<FileItem*>(item);
 
-		contextMenu.addAction(
-			QIcon(embed::getIconPixmap("folder")),
-			tr("Open containing folder"),
-			[=, this]{ openContainingFolder(file); }
-		);
+   QMenu contextMenu( this );
 
-		auto songEditorHeader = new QAction(tr("Song Editor"), nullptr);
-		songEditorHeader->setDisabled(true);
-		contextMenu.addAction( songEditorHeader );
-		contextMenu.addActions( getContextActions(file, true) );
+   auto dir = dynamic_cast<Directory*>(item); // TODO: this might not be a great way to check if it's a directory
 
-		auto patternEditorHeader = new QAction(tr("Pattern Editor"), nullptr);
-		patternEditorHeader->setDisabled(true);
-		contextMenu.addAction(patternEditorHeader);
-		contextMenu.addActions( getContextActions(file, false) );
 
-		// We should only show the menu if it contains items
-		if (!contextMenu.isEmpty()) { contextMenu.exec( e->globalPos() ); }
-	}
+   if( file != nullptr)
+   {
+	   if (file->isTrack()) {
+		   contextMenu.addAction(
+			   tr( "Send to active instrument-track" ),
+			   [=, this]{ sendToActiveInstrumentTrack(file); }
+		   );
+
+		   contextMenu.addSeparator();
+	   }
+
+	   contextMenu.addAction(
+		   QIcon(embed::getIconPixmap("folder")),
+
+		   tr("Show in") + " " +fileManager,
+		   [=, this]{ openContainingFolder(file); }
+	   );
+
+
+	   auto songEditorHeader = new QAction(tr("Song Editor"), nullptr);
+	   songEditorHeader->setDisabled(true);
+	   contextMenu.addAction( songEditorHeader );
+	   contextMenu.addActions( getContextActions(file, true) );
+
+	   auto patternEditorHeader = new QAction(tr("Pattern Editor"), nullptr);
+	   patternEditorHeader->setDisabled(true);
+	   contextMenu.addAction(patternEditorHeader);
+	   contextMenu.addActions( getContextActions(file, false) );
+
+   } else if (dir) {
+
+	   contextMenu.addAction(
+		   QIcon(embed::getIconPixmap("folder")),
+		   tr("Open in") + " " + fileManager,
```

Comment:
For better translation in some languages, this would be better:
```suggestion
		   tr("Open in %1").arg(fileManager),
```